### ISSUE TITLE
Automatic unit compatibility checking

### ISSUE THREAD

**Issue by kyleabeauchamp** [openmm/openmm#204]
```
timestep = 2.0 / u.picoseconds
integrator = mm.VerletIntegrator(timestep)

In [45]: timestep
Out[45]: Quantity(value=2.0, unit=/picosecond)

In [46]: integrator.getStepSize()
Out[46]: Quantity(value=2.0, unit=picosecond)

```

To me, this looks like the unit conversion is trying to be smart and automatically invert per-picoseconds into picoseconds.  However, it seems to be failing.  

IMHO, this sort of automatic inversion should not even be attempted without a warning or error.  

Thoughts?

**Comment by jchodera** [comment]
Let's mark this as a potential "enhancement".  We can try the `accepts_compatible_units` unit checking soon.

**Comment by jchodera** [comment]
Maybe rename the issue "Automatic unit compatibility checking"?

**Comment by peastman** [comment]
This is fixed by #3010.
